src/olympia/accounts/views.py

```diff
@@ -397,8 +398,22 @@ def get(self, request, user, identity, next_path):
                 user = register_user(identity)
             else:
                 reregister_user(user)
+            if not is_safe_url(next_path, request):
+                next_path = None
+            # If we just reverse() directly, we'd use a prefixer instance
+            # initialized from the current view, which would not contain the
+            # app information since it's a generic callback, the same for
+            # everyone. To ensure the user stays on the app/locale they were
+            # on, we extract that information from the next_path if present
+            # and set locale/app on the prefixer instance that reverse() will
+            # use automatically.
+            prefixer = get_url_prefix()
```

Comment:
It seems like the logic on lines 410-414 are only needed if `next_path` is truthy, and we set it to `None` above after the `if` on line 401, so should all this logic also be gated on that `if` statement?